Add local csv loading, empty state message, prettier formatting, const

notitatall opened the pull request:
**New**
Added a code path for loading a local csv file

Added ability for users to specify a threshold (number of days) that count as "inactivity"

Added a message for when a dry run shows we'll deprovision no users (this will likely be useful now that a user can adjust the threshold).

**aesthetic** 💅
Formatted the code with Prettier, switch our `var`s to `const`s, and also switched it all to smaller arrow functions.

Commit:
Add local csv loading, empty state message, prettier formatting, const

alexdavies74 commented on index.js:
```diff
@@ -1,106 +1,129 @@
+    );
+  }
+
+  if (process.argv[6] !== "action") {
```
I regret not using a command line options module now. This edit will break existing command lines, because the number of days is mandatory, and appears before the "action", so it'll try to use "action" as the number of days.

If you feel like pulling off the band aid and switching to --domain_id= --action etc, I'd support that, and we should do it now, before changing the command line format multiple times.

alexdavies74 commented on index.js:
```diff
@@ -1,106 +1,129 @@
+  const deprovision = deprovisionFunc();
+
+  // Decide whether or not this is a file path or a url
+  const csvData = process.argv[3].includes("https://")
```
I'd pull process.argv[3] out as a variable, specifically because its name would serve as documentation.

alexdavies74 reviewed (approved):
For future diffs, I'd have preferred the const and prettier in a separate diff, to make reviewing the meaty changes easier.

Did you try it out? I'm not in the habit of writing automated tests for things where the complexity lies mostly in using APIs, but am always nervous that they will grow and be used for more than I originally expected, and regret that decision. There should be a time when we say "it's complex enough for tests now". Do you have an opinion about that time? If you think it's now, feel free to add tests, and feel free to tap me to do half of them.

Commit:
update deps, add commander, separate fn concerns, unit tests

notitatall commented:
K! You made some good points there, so decided to rip the bandaid off. I've added the commander library so we that we have options rather than just parsing each argument manually. This means that order no longer matters.

This required a fair amount of refactoring that retrospect should have been done in separate commits (sorry about that, was moving quickly).

I had tested this before and it was working for local csvs (and I suppose I presumed it was working for unsigned remote csvs). I don't think generated a signed csv or local csv and then running the script against that is a particularly pleasant way of testing the functions in this script moving forward so I added Jest and wrote some unit tests in this latest commit. These tests certainly don't cover every case yet, but they ought to give us enough coverage reasonably say the script works for typical cases.

I separated out some of the concerns in the script's functions to make them more testable, I believe those to be good changes and they make the flow of user inputs through the script a little more linear.

Commit:
update documentation in readme.md
